### DISCUSSION TITLE
Really a nice project, but I have some questions

### DISCUSSION THREAD

**Discussion by DanieleNeo** [Trunks1982/Daikin-Rotex-HPSU-CAN#63] [General]
Hi,

as per the subject, a really nice project, congratulations. I stumbled upon it by chance, and I'm happy about it, I'll explain the background in a moment.

I am the owner of a Daikin Rotex HPSU Compact 508 H/C BIV heat pump purchased in 2014 and never put into service due to a series of issues for which I won't bore you here. This summer I installed it and now I'm waiting for it to be started, and because Daikin doesn't support it too much since it's an old hardware and I didn't consider it convenient to resort to the deprecated proprietary solutions for monitoring and remote control, I took action to see if it wasn't possible to integrate the heat pump with home automation, being familiar with Home Assistant and software developing in general.

I then discovered ESPAltherma, implementing a modification to the code to manage the activation of cooling as well as heating, in addition to creating a board with 4N35 to drive the thermostat (connector J16, here my discussion about it https://github.com/raomin/ESPAltherma/discussions/467). But ESPAltherma does not return some values ​​to calculate the COP, for example the flow rate, and in general the serial connection allows you to retrieve information but not change it, which can be done via CAN bus and P1P2. I must say that being able to build from source directly from EspHOME is really fantastic, and therefore in my case from HomeAssistant, I have to work a little on the localization, unfortunately German is not my strong point ;)

So my questions:
- via CAN bus is it not possible to act directly on the thermostat, in heating/cooling mode? Because somewhere I saw that you suggest using relays and J16, which I already do with ESPAltherma
- are CAN bus and P1P2 equivalent in terms of functions that can operate or are there differences?

Thanks,
D.

**Comment by wrfz** [comment]
I don't have a thermostat, so I can't say much about it. As far as I understood in the telegram group,
it actually has nothing to do with the HPSU and CAN. You simply create an ESPhome switch and connect it to your ESP.

As far as the CAN project is concerned, you can read out almost all the values that you see in your display with this CAN project and also change all the settings. In addition to the usual settings such as operating mode, room setpoint, hot water setpoint, you can also configure special settings such as the minimum or maximum output of the circulation pump and much more.

![grafik](https://github.com/user-attachments/assets/43725d6d-73c1-4cc2-9fda-9bf002a036f2)
![grafik](https://github.com/user-attachments/assets/ae22830d-1417-496e-b2ba-d2d311d098c7)

I can't say anything about P1P2 because I don't have it.
You can try your luck in the Telegram group. Most of the communication there is in German, but there are also a few people who prefer English.

**Comment by DanieleNeo** [comment]
Hi @wrfz , thanks for your reply. Yes I know this approach, it's the same I used with EspAltherma, even if I use two 4N35 optocouplers instead of relays (to save power, I power everything from the 5V of the serial interface), but using the serial port is mandatory, there you only read and not write. I know that with P1P2 (which I also don't have on my device) the thing is feasible, I was wondering if it wasn't also with the CAN, RoCon U1 and RoCon M1 can't start the heating and cooling remotely? if they can it should be feasible, you would have a completely digital solution without resorting to J16. Simple curiosity.

D.

**Reply by Trunks1982** [reply]
Rocon u1 ist im prinzip das selbe wie die Bedieneinheit direkt an der Heizung. Wie schon erwähnt über unsere Lösung sind keine Grenzen gesetzt die Heizung zu steuern.

**Reply by DanieleNeo** [reply]
@Trunks1982, consider that I am ignorant, if Rocon u1 is essentially a remote of what it does in the device then you answered me, through CAN it is not possible to operate the thermostat, and therefore we go to J16. A great project indeed, I'll start working a bit on the localization for personal use ;)

D.

**Reply by Trunks1982** [reply]
Das stimmt das man per canbus nicht j16 ansteuern kann.Muss mann aber auch nicht zwingend.Dafür kann man ja den Betriebsmodus steuern.Hab ich auch schon alles gemacht.ist eigentlich nichts anderes als j16 anzusteuern.
